gateway/ln-gateway/src/rpc/mod.rs

```diff
@@ -50,11 +51,31 @@ pub struct DepositAddressPayload {
 #[derive(Debug, Serialize, Deserialize, Clone)]
 pub struct WithdrawPayload {
     pub federation_id: FederationId,
-    #[serde(with = "bitcoin::util::amount::serde::as_sat")]
-    pub amount: bitcoin::Amount,
+    pub amount: AmountOrAll,
     pub address: Address,
 }
 
+#[derive(Debug, Serialize, Deserialize, Clone)]
+#[serde(untagged)]
+pub enum AmountOrAll {
```

Comment:
Maybe call it `BitcoinAmountOrAll`? Just looking at the struct name I would assume this was a fedimint amount. Also, could you add a doc string to this struct https://x.com/dpc_pw/status/1722457120350589305?s=20